python/cudf/cudf/pandas/profiler.py

```diff
@@ -163,6 +163,15 @@ def _tracefunc(self, frame, event, arg):
             ):
                 func_name = self.get_namespaced_function_name(func_obj)
                 self._call_stack.append((func_name, time.perf_counter()))
+        elif (
+            event == "call"
+            and frame.f_code.co_name != "_fast_slow_function_call"
+            # TODO: Add a condition to filter other functions in same cell as %%cudf.pandas.profile
```

Comment:
Right now all non `_fast_slow_function_calls` are timed, but we need to be more restrictive than that. But as of now, I'm not sure the best way to do that

Reply:
> we need to be more restrictive than that.

Why is that? What would we want to exclude?

Reply:
I think because we should only show time taken by functions called in the`%%cudf.pandas.profile` cell. In this [example](https://github.com/rapidsai/cudf/pull/16522#issuecomment-2278578693), the toal time taken is 1.485. And 1 sec is eaten up by the `time.sleep` call.